egs/swbd/s5c/local/nnet3/run_discriminative.sh

```diff
@@ -0,0 +1,221 @@
+#!/bin/bash
+
+set -e 
+set -o pipefail
+
```

Comment:
Vimal, I don't want to perpetuate this approach of encoding all the details of the configuration into the filename.  That way lies madness.  Can you please just let the directory of the experiment and its egs be a user-configurable value, and if users want to run multiple versions of the experiment, let them change it?

Also, bear in mind that run.sh scripts are supposed to be example scripts, they do not have to support extensive options (only the --stage and --train-stage options are important), so any configurable values do not have to be configurable from the command line.  If a variable is used only once it may be easier to just use the value directly rather than defining a variable.